### DISCUSSION TITLE
unable to capture the response and save it as variable

### DISCUSSION THREAD

**Discussion by arunav1101** [artilleryio/artillery#1627] [Artillery]
I am unable to capture the response and save it . The API is returning response as Text and when i am trying capture ,it is not storing the the variable.

        url: "/login/user"
        headers:
          Authorization: "xxxx"
        capture:
          - response: "$"
            as: token 
        expect:
          - statusCode: 200
    - log: "{{token}} " 

DEBUG:

GET /login/user
  ok statusCode 200 
2022-10-19T18:28:51.437Z http:response {
  "date": "Wed, 19 Oct 2022 18:28:51 GMT",
  "server": "Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips",
  "x-powered-by": "Express",
  "access-control-allow-origin": "*",
  "content-type": "text/html; charset=utf-8",
  "content-length": "2189",
  "keep-alive": "timeout=5, max=100",
  "connection": "Keep-Alive"
}

2022-10-19T18:28:51.437Z http:response "\"eyJhbGciOiJIUzI1NiIsInxxxxxxxxxxxxx\""
2022-10-19T18:28:51.438Z http captures and matches:
2022-10-19T18:28:51.439Z http {}
2022-10-19T18:28:51.440Z http { token: { value: '', strict: undefined, failed: true } }

Can someone help me on this?

**Comment by hassy** [comment]
hi @arunav1101 - you need to set capture attribute to `json` if you're using JSONPath. The response in your example is not JSON, so if you want to capture the entire response, use a regular expression:

https://www.artillery.io/docs/guides/guides/http-reference#syntax
